### PR TITLE
Fixes in runtime v3

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: client-runtime/src/main/java/com/microsoft/rest/v3/http/HttpClientConfiguration.java [hunk 1]
```diff
@@ -6,42 +6,29 @@
...
     }
 
-    /**
-     * Sets the duration in sec to keep the connection alive in available pool before closing it.
-     *
-     * @param duration duration in seconds
-     * @return HttpClientConfiguration
-     */
-    @Beta(since = "2.0.0")
-    public HttpClientConfiguration withIdleConnectionKeepAliveDurationInSec(long duration) {
```
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v3/http/HttpClientConfiguration.java]: This was deleted by mistake. Add back

File: client-runtime/src/main/java/com/microsoft/rest/v3/http/NettyClient.java [hunk 2]
```diff
@@ -26,13 +25,12 @@
...
 import java.util.Objects;
 import java.util.function.BiFunction;
-import java.util.function.Function;
 
 /**
  * An HttpClient that is implemented using Netty.
  */
 public final class NettyClient extends HttpClient {
-    private final HttpClientConfiguration configuration;
+    private reactor.netty.http.client.HttpClient httpClient;
```
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v3/http/NettyClient.java]: final

File: client-runtime/src/main/java/com/microsoft/rest/v3/http/ProxyOptions.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    /**
+     * @return the type of the proxy.
+     */
+    public Type type() {
+        return type;
+    }
+
+    public enum Type {
```
Comment (jianghaolu) [inline client-runtime/src/main/java/com/microsoft/rest/v3/http/ProxyOptions.java]: javadocs

File: client-runtime/src/main/java/com/microsoft/rest/v3/http/NettyClient.java (hunk 2 above)
Comment (jianghaolu) [reply client-runtime/src/main/java/com/microsoft/rest/v3/http/NettyClient.java]: Yuck. Can't do. Reactor creates a new instance when configuring the TCP configurations.

**Comment by jianghaolu** [comment]
@anuchandy Please also review https://github.com/Azure/autorest-clientruntime-for-java/pull/540/commits/6fd8f15f1a82a862da84458e3412c22c896ccf48

**Comment by jianghaolu** [comment]
There are still more LRO errors but I'll finalize the PR with these fixes for now.

**Comment by anuchandy** [comment]
@jianghaolu thanks, this LGTM.

**Pull request merged**